msal/src/main/java/com/microsoft/identity/client/statemachine/states/BaseStates.kt

```diff
@@ -0,0 +1,286 @@
+import com.microsoft.identity.client.NativeAuthPublicClientApplication
+import com.microsoft.identity.client.NativeAuthPublicClientApplicationConfiguration
+import com.microsoft.identity.client.PublicClientApplication
+import com.microsoft.identity.client.exception.MsalClientException
+import com.microsoft.identity.client.exception.MsalException
+import com.microsoft.identity.client.internal.CommandParametersAdapter
+import com.microsoft.identity.client.statemachine.results.SignOutResult
+import com.microsoft.identity.common.internal.commands.AcquireTokenNoFixedScopesCommand
+import com.microsoft.identity.common.internal.commands.RemoveCurrentAccountCommand
+import com.microsoft.identity.common.internal.controllers.LocalMSALController
+import com.microsoft.identity.common.internal.controllers.NativeAuthMsalController
+import com.microsoft.identity.common.java.commands.CommandCallback
+import com.microsoft.identity.common.java.controllers.CommandDispatcher
+import com.microsoft.identity.common.java.controllers.ExceptionAdapter
+import com.microsoft.identity.common.java.dto.AccountRecord
+import com.microsoft.identity.common.java.eststelemetry.PublicApiId
+import com.microsoft.identity.common.java.exception.BaseException
+import com.microsoft.identity.common.java.exception.ServiceException
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.logging.Logger
+import com.microsoft.identity.common.java.logging.Logger.LogLevel
+import com.microsoft.identity.common.java.result.ILocalAuthenticationResult
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import java.io.Serializable
+
+sealed interface State
+
+abstract class BaseState(internal open val flowToken: String?)
+
+/**
+ *  AcccountResult returned as part of a successful completion of sign in flow [com.microsoft.identity.client.statemachine.results.SignInResult.Complete].
+ */
+class AccountResult private constructor(
+    private val account: IAccount,
+    private val config: NativeAuthPublicClientApplicationConfiguration
+) : Serializable {
+
+    companion object {
+
+        private val TAG = NativeAuthPublicClientApplication::class.java.simpleName
+
+        fun createFromAuthenticationResult(
+            authenticationResult: IAuthenticationResult,
+            config: NativeAuthPublicClientApplicationConfiguration
+        ): AccountResult {
+            return AccountResult(
+                account = authenticationResult.account,
+                config = config
+            )
+        }
+
+        fun createFromAccountResult(
+            account: IAccount,
+            config: NativeAuthPublicClientApplicationConfiguration
+        ): AccountResult {
+            return AccountResult(
+                account = account,
+                config = config
+            )
+        }
+    }
+
+    interface SignOutCallback : Callback<SignOutResult>
```

Comment:
Javadoc

Reply:
Also what's inside this interface?  Seems to be empty

Reply:
I mean I don't see any curly braces...so not sure if the below methods are in this interface or on the AccountResult class or perhaps on the parent interface i.e. BaseStates?